youtube_dl/extractor/youku.py

```diff
@@ -22,9 +22,14 @@ class YoukuIE(InfoExtractor):
     _VALID_URL = r'''(?x)
         (?:
             https?://(
-                (?:v|player)\.youku\.com/(?:v_show/id_|player\.php/sid/)|
-                video\.tudou\.com/v/)|
-            youku:)
+                v\.youku\.com/v_show/id_|
+                player\.youku\.com/player\.php/sid/|
+                player\.youku\.com/embed/|
+                cloud\.youku\.com/services/sharev\?vid=|
+                video\.tudou\.com/v/
```

Comment:
This no longer matches http://v.youku.com/player.php/sid/XNDgyMDQ2NTQw/v.swf and http://player.youku.com/v_show/id_XMTc1ODE5Njcy.html.

Reply:
In fact those are invalid, you can check the links and will find 404. The original regex is overmatching.

Reply:
I'm well aware of that.

Reply:
@dstftw So, what should I do to get this fixed?

Reply:
Make it match URLs from the first post obviously.

Reply:
All valid URLs are already listed in `_TESTS`, and they are all matched by the patched regex.
IMHO, URLs from the first post is invalid so they should not be testcases, and there is no reason to support URLs not in `_TESTS`.